rflx/generator/common.py

```diff
@@ -118,14 +120,10 @@ def byte_aggregate(aggregate: Aggregate) -> Aggregate:
                         ),
                         aggregate,
                     )
-                return Call("Equal", [Variable("Ctx"), Variable(field.affixed_name), aggregate])
-
-        literals = [
-            l
-            for t in message.types.values()
-            if isinstance(t, Enumeration)
-            for l in t.literals.keys()
-        ]
```

Comment:
Why are literals not a thing anymore?

Reply:
The code using `literals` had no effect. Literals are unconditionally substituted due to `substitution_facts`. The code added to `substitution` in 0f30f7be8c42da77321c5f4afa482c0ccf95d196 is just an optimization to improve provability. It prevents unnecessary type conversions in some cases. This was needed because `gnatprove` seems to have a hard time proving equality after conversion to our `U64` type (at least in some cases).

Before removing the code, I tried to fix it. But the correct handling of enumeration literals is not that easy and would take much more time (one reason is the different handling of `Always_Valid` enums). That would also only cover special cases. I doubt that we are able to prevent all unnecessary type conversions, as we have no static typing in the specification. So I'm not sure if trying to prevent some more unnecessary type conversions is worth the effort. What do you think?

Reply:
It agree that it's not worth the effort (for a partial solution, at least). Also, the removal has actually improved the readability of the code. Ultimately, we will need #87 to get this right. Thanks for the explanation.